data_engineering.qmd

```diff
@@ -1,121 +1,526 @@
+
+**[[lakeFS]{.underline}](https://docs.lakefs.io/):** It is an open-source tool that supports the data version control on data lakes. It supports many git-like operations such as branching and merging of data, as well as reverting to previous versions of the data. It also has a unique UI feature which allows exploration and management of data much easier.
+
+**[[Git LFS]{.underline}](https://git-lfs.com/):** It is useful for data version control on smaller sized datasets. It uses Git's inbuilt branching and merging features, but is limited in terms of tracking metrics, reverting to previous versions or integration with data lakes.
 
-- Version Control Systems 
-- Metadata 
 
 ## Optimizing Data for Embedded AI
 
-Explanation: This section concentrates on optimization techniques specifically suited for embedded systems, focusing on strategies to reduce data volume and enhance storage and retrieval efficiency, crucial for resource-constrained embedded environments.
 
-- Low-Resource Data Challenges
-- Data Reduction Techniques 
-- Optimizing Data Storage and Retrieval 
+Creators working on embedded systems may have unusual priorities when cleaning their datasets. On the one hand, models may be developed for unusually specific use cases, requiring heavy filtering of datasets. While other natural language models may be capable of turning any speech to text, a model for an embedded system may be focused on a single limited task, such as detecting a keyword. As a result, creators may aggressively filter out large amounts of data because they do not address the task of interest. Additionally, an embedded AI system may be tied to specific hardware devices or environments. For example, a video model may need to process images from a single type of camera, which will only be mounted on doorbells in residential neighborhoods. In this scenario, creators may discard images if they came from a different kind of camera, show the wrong type of scenery, or were taken from the wrong height or angle.
+
+On the other hand, embedded AI systems are often expected to provide especially accurate performance in unpredictable real-world settings. As a result, creators may design datasets specifically to represent variations in potential inputs and promote model robustness. As a result, they may define a narrow scope for their project but then aim for deep coverage within those bounds. For example, creators of the doorbell model mentioned above might try to cover variations in data arising from:
+
+- Geographically, socially and architecturally diverse neighborhoods 
+- Different types of artificial and natural lighting
+- Different seasons and weather conditions
+- Obstructions (e.g. raindrops or delivery boxes obscuring the camera’s view)
```

Comment:
Maybe also mention that the type of sensors/hardware used to collect this data should be taken into consideration when collecting data as well (ex: resolution limitations, any pre/post-processing, etc.)

Reply:
to add to this comment, it may also mention the installation options for final hardware (different angles, installation heights, etc.)